backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -362,6 +364,17 @@ private void unlockDisk() {
         imageDao.updateStatusOfImagesByImageGroupId(getParameters().getImageGroupID(), ImageStatus.OK);
     }
 
+    private void releaseSnapshotLock() {
+        EngineLock removeSnapshotLock = createEngineLockForSnapshotRemove();
+        lockManager.releaseLock(removeSnapshotLock);
+    }
+
+    private EngineLock createEngineLockForSnapshotRemove() {
+        return new EngineLock(
+                getExclusiveLocksForSnapshotRemove(),
```

Comment:
I don't think this is right, we don't need the disk lock, what we actually need is an exclusive lock on the VM and a shared lock is taken.

(To be clear, this was an existing problem introduced in https://github.com/mkemel/ovirt-engine/commit/b5d549933501ec113d44d7becbe926ed29eff9e3)